frontend/javascripts/admin/organization/upgrade_plan_modal.tsx

```diff
@@ -0,0 +1,196 @@
+import React, { useRef } from "react";
+import { Divider, InputNumber, Modal } from "antd";
+import moment from "moment";
+import {
+  DatabaseOutlined,
+  FieldTimeOutlined,
+  RocketOutlined,
+  UserAddOutlined,
+} from "@ant-design/icons";
+import { APIOrganization } from "types/api_flow_types";
+import { formatDateInLocalTimeZone } from "components/formatted_date";
+import {
+  sendExtendPricingPlanEmail,
+  sendUpgradePricingPlanEmail,
+  sendUpgradePricingPlanStorageEmail,
+  sendUpgradePricingPlanUserEmail,
+} from "admin/admin_rest_api";
+import { powerPlanFeatures, teamPlanFeatures } from "./pricing_plan_utils";
+import { PricingPlanEnum } from "./organization_edit_view";
+
+function extendPricingPlan(organization: APIOrganization) {
+  const extendedDate = moment(organization.paidUntil).add(1, "year");
+
+  Modal.confirm({
+    title: "Extend Current Plan",
+    okText: "Request Extension",
+    onOk: sendExtendPricingPlanEmail,
+    icon: <FieldTimeOutlined style={{ color: "var(--ant-primary-color)" }} />,
+    width: 1000,
+    content: (
+      <div
+        style={{
+          background:
+            "right -150px / 50% no-repeat url(/assets/images/pricing/background_neurons.png)",
+        }}
+      >
+        <p style={{ marginRight: "30%" }}>
+          Extend your plan now for uninterrupted access to webKnossos. Expired plans will be
+          downgraded to the Free plan and you might lose access to some webKnossos features and see
+          restrictions on the number of permitted user accounts and your included storage space
+          quota.
+        </p>
+        <p>
+          Your current plan is paid until:{" "}
+          {formatDateInLocalTimeZone(organization.paidUntil, "YYYY-MM-DD")}
+        </p>
+        <p>Buy extension until: {extendedDate.format("YYYY-MM-DD")}</p>
+        <Divider style={{ marginTop: 40 }} />
+        <p style={{ color: "#aaa", fontSize: 12 }}>
+          Requesting an extension will send an email to the webKnossos sales team to extend your
+          current plan. We typically respond within one business day. webKnossos plans are billed
+          annually. See our <a href="https://webknossos.org/faq">FAQ</a> for more information.
+        </p>
+      </div>
+    ),
+  });
+}
+
+function upgradeUserQuota() {
+  const userInputRef = useRef<HTMLInputElement | null>(null);
```

Comment:
nice solution :+1:

Reply:
argh, this is still problematic because it's not a react component. I forgot about that when reading it. however, the button is still broken :S

Reply:
Fixed